src/named-node.ts

```diff
@@ -1,54 +1,58 @@
-'use strict'
 import ClassOrder from './class-order'
 import Node from './node-internal'
+import { ValueType } from './types';
 
 /**
- * @class NamedNode
- * @extends Node
- */
+* A named (IRI) RDF node
+*/
 export default class NamedNode extends Node {
+  static termType: string;
+
   /**
-   * @constructor
-   * @param iri {String}
+   * Initializes this node
+   * @param iri The IRI for this node
    */
-  constructor (iri) {
+  constructor (iri: NamedNode | string) {
     super()
     this.termType = NamedNode.termType
 
-    if (iri && iri.termType === NamedNode.termType) {  // param is a named node
-      iri = iri.value
+    if (iri && (iri as NamedNode).termType === NamedNode.termType) {  // param is a named node
```

Comment:
Tip: create a [user-defined type guard](https://www.typescriptlang.org/docs/handbook/advanced-types.html#user-defined-type-guards) (e.g. `isNamedNode()`), then there's no need to do the cast inside the `if` block. (Or even after it, I guess, since `string` is the complementary type.)